papers/ring_vrf/zk_cont.tex

```diff
@@ -1,34 +1,101 @@
-
-\section{Zero-knowledge continuations}
+\section{Zero-knowledge Continuations}
 \label{sec:rvrf_cont}
 
-We now develop our zero-knowledge continuation technique, called 
-{\em special(ized) Groth16} or \SpecialG, and
- build ring VRFs with fast amortized prover time.
-Anonymity demands we finalize the amortized ``continuation'' multiple
-times, with each time being unlinkable to the others, meaning our rVRF
-stays zero-knowledge even with the continuation being reused.
-
-Our zero-knowledge continuation $\SpecialG_\rel$ consists of three algorithms:
+\noindent In the following, we describe a NIZK for a relation $\rel$ where
+$$\rel = \{(\bary, \barz; \barx, \baromega, \baromegap):  (\bary, \barx; \baromega) \in \relone, (\barz, \barx; \baromegap) \in \reltwo \},$$
+and $\relone$, $\reltwo$ are some NP relations. Our NIZK is designed to efficiently re-prove membership for relation $\relone$
```

Comment:
\barx is a witness in \rel but not in \relone and \reltwo. Is this correct?

Reply:
Yes. It's the witness data they have in common, so it is part of the witness for the whole relation (we should be ZK for it), but the whole relation needs to check that it is the same between the two sub-relations so it shopuld be in their inputs.